### PR TITLE
Two minor bug fixes for Spark Web UI

### PR DESCRIPTION
**Pull request by JoshRosen**
This pull request fixes two minor issues in the Spark Web UI:
- [SPARK-630](https://spark-project.atlassian.net/browse/SPARK-630): App details page lists finished/killed executors as RUNNING.
- [SPARK-629](https://spark-project.atlassian.net/browse/SPARK-629): App details page has strange value for number of cores.

I also fixed a minor HTML issue where an `<a>` tag was closed with `</href>`.

### CODE REVIEW

File: core/src/main/scala/spark/deploy/ApplicationDescription.scala [hunk 1]
```diff
@@ -2,7 +2,7 @@ package spark.deploy
 
 private[spark] class ApplicationDescription(
     val name: String,
-    val cores: Int,
+    val maxCores: Int, /* Integer.MAX_VALUE denotes an unlimited number of cores */
```
Comment (JoshRosen) [inline core/src/main/scala/spark/deploy/ApplicationDescription.scala:5]: The issue with the weird number of cores in the web UI seems to stem from the fact that this seems to be the _maximum number of_ cores that the application may use, but `Integer.MAX_VALUE` wasn't being consistently treated as "unlimited" in the web UI code.  I think that renaming this to `maxCores` makes the code clearer.

File: core/src/main/scala/spark/deploy/JsonProtocol.scala [hunk 2]
```diff
@@ -26,7 +26,7 @@ private[spark] object JsonProtocol extends DefaultJsonProtocol {
       "starttime" -> JsNumber(obj.startTime),
       "id" -> JsString(obj.id),
       "name" -> JsString(obj.desc.name),
-      "cores" -> JsNumber(obj.desc.cores),
+      "cores" -> JsNumber(obj.desc.maxCores),
```
Comment (JoshRosen) [inline core/src/main/scala/spark/deploy/JsonProtocol.scala:29]: Although I renamed this to `maxCores` in the internal API, I left the JSON API unmodified.

File: core/src/main/scala/spark/deploy/master/Master.scala [hunk 3]
```diff
@@ -275,6 +275,7 @@ private[spark] class Master(host: String, port: Int, webUiPort: Int) extends Act
       for (exec <- app.executors.values) {
         exec.worker.removeExecutor(exec)
         exec.worker.actor ! KillExecutor(exec.application.id, exec.id)
+        exec.state = ExecutorState.KILLED
```
Comment (JoshRosen) [inline core/src/main/scala/spark/deploy/master/Master.scala:278]: `exec.state` is ordinarily updated in the `ExecutorStateChanged` message handler, but that isn't fired when we kill the executors.  As a result, killed executors were still marked as `RUNNING` in the master Web UI.

**Comment by mateiz** [comment]
Thanks for the fixes, Josh.

**Pull request merged**